### ISSUE TITLE
Pass arguments to the subprocesses explicitly (support non-fork start method in OSX with Pyhon 3.8+)

### ISSUE THREAD

**Issue by eldipa** [byexamples/byexample#118]
**Describe the feature you'd like**
We use `multiprocesing` for running the tests in parallel. When the lib uses a `fork` strategy to spawn the subprocesses, global objects of `byexample` are shared (copied). This makes easy to pass info to a child but it works only in linux (and in osx with python 3.7 or inferior)

The idea is to pass the info explicitly so we can use other mechanisms to spawn subprocesses.

**Describe alternatives you've considered (optional)**
The most simple way is to pickle the arguments.

**Additional context (optional)**
See #116

**Comment by eldipa** [comment]
Unfortunately this is much complex than it looks. `byexample` passes global objects to its subprocesses via a `fork`with a copy bit-by-bit and other methods like `spawn` or `forkserver` would require a serialization with `pickle`.

The problem is that the objects are non-trivial and I found several attributes that cannot be pickled:
 - `Argparse` registers a `identity` function which it is not global
 - `cfg['output']` used by `Progress` points to `sys.stdout`
 - `Harvester` and `Concern` have a `dict_value` (the result of `aDict.values()`)

and they may be others.

As a workaround for macOS users we can force the `fork` method (other will not work so no problem here) #125

**Comment by eldipa** [comment]
Other alternatives were explored to no relay on `multiprocessing` but all of the have some drawbacks:
 - use threads instead of `multiprocessing`: implicit sharing is free but problems with the GIL may appear. Even if the GIL is released during the regex matching (one of the most CPU intensive part of `byexample`), running multiple threads with one of them doing a `fork` is not a good thing to do in OSX. This can happen when we run an interpreter using `pexpect`
 - use `async` code: we should put all in a single threads and use ` async` for the blocking calls. Unfortunately, the regex matching is synchronous and part of the `pexpect` API is still synchronous.

**Comment by eldipa** [comment]
In #131 we moved the concurrency model from `multiprocessing` to `threading`. In this process we were able to "move" `cfg['output']`, `Harvester`, `Concern` and other objects from the main thread to the worker threads.

This "move" was done by recreating this objects from very simple, plain arguments. This makes the code thread safe but also makes the pickling of the arguments possible.

There are still things that need to be resolved:
 - `Argparse` registers an `identity` function which it is not global
 - `init_byexample` creates a `namedtuple` which cannot be pickled. In `Python 3.6` we could replace it be a `Namespace` (shareable among threads/processes) but we need to find a way to make it immutable because it will be shared among the workers and therefore we don't want to open the opportunity to RC.
